hypothesis-python/src/hypothesis/internal/observability.py

```diff
@@ -19,6 +19,7 @@
 from hypothesis.internal.conjecture.data import ConjectureData, Status
 
 TESTCASE_CALLBACKS: List[Callable[[dict], None]] = []
+OBSERVABILITY_COLLECT_COVERAGE = True
```

Comment:
Let's set this as `OBSERVABILITY_COLLECT_COVERAGE = "HYPOTHESIS_EXPERIMENTAL_OBSERVABILITY_NOCOVER" not in os.environ`, put it just above `if "HYPOTHESIS_EXPERIMENTAL_OBSERVABILITY" in os.environ:`, and change that condition to `if "HYPOTHESIS_EXPERIMENTAL_OBSERVABILITY" in os.environ or OBSERVABILITY_COLLECT_COVERAGE is False:`.

That way setting _only_ `HYPOTHESIS_EXPERIMENTAL_OBSERVABILITY_NOCOVER` will enable observability in nocover mode.

The last thing I'd do is emit a warning in nocover mode on Python >= 3.12, because the `sys.monitoring` system is expected to have much lower overhead and users might not know that.  Maybe worth a sentence in the docs too?

Reply:
OK, I think I did all of this correctly!